osu.Game/Scoring/Legacy/LegacyScoreDecoder.cs

```diff
@@ -27,6 +27,11 @@ namespace osu.Game.Scoring.Legacy
 {
     public abstract class LegacyScoreDecoder
     {
+        /// <summary>
+        /// The decoded "IsPerfect" value. This isn't used by osu!lazer.
+        /// </summary>
+        public bool DecodedPerfectValue { get; private set; }
```

Comment:
This feels very ugly to do just for a test. To the point that I'm having very big trouble accepting it.

I think I'd rather just do this in the test. Encode the score as normal, and then directly use `SerializationReader` to skip the cruft and read this one flag as desired. Compare: https://github.com/ppy/osu/blob/498948f17d4e2cd3dcc205502053402a2721777b/osu.Game.Tests/Beatmaps/Formats/LegacyScoreDecoderTest.cs#L264-L283 (although that's in the opposite direction).

Reply:
Had a feeling that would happen. Done.